src/components/Catalyst/Step4.js

```diff
@@ -174,7 +174,7 @@ const Step4 = ({
       }
     },
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    [],
+    [password],
```

Comment:
what happens if you add all the dependencies?

Reply:
@wolverineks , what other dependencies are critical inside that callback? 🤔 I couldn't figure anything except the password. I see it's using the setters from other declared states (like `setGeneratingTransaction` and `setErrorData`), but never actually refers to the values.

Reply:
including all the dependencies should be the default mindset, and excluding any deserves scrutiny
my preference is to figure out why something is excluded, and try to solve the problem before it gets to `useEffect`
that might mean defining functions in `useCallback` or defining them directly in `useEffect`

Reply:
@wolverineks , good point! This PR was in response to a specific found issue, but I would have no objections to have another PR from you improving the code wherever necessary 👌